service/service_windows.go

```diff
@@ -0,0 +1,146 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build windows
+
+package service
+
+import (
+	"fmt"
+	"syscall"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap/zapcore"
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/eventlog"
+)
+
+type WindowsService struct {
+	params Parameters
+	app    *Application
+}
+
+func NewWindowsService(params Parameters) *WindowsService {
+	return &WindowsService{params: params}
+}
+
+func (s *WindowsService) Execute(args []string, requests <-chan svc.ChangeRequest, changes chan<- svc.Status) (bool, uint32) {
+	// The first argument supplied to service.Execute is the service name. If this is
+	// not provided for some reason, raise a relevant error to the system event log
+	if len(args) == 0 {
+		return false, 1213 // 1213: ERROR_INVALID_SERVICENAME
+	}
+
+	elog, err := openEventLog(args[0])
+	if err != nil {
+		return false, 1501 // 1501: ERROR_EVENTLOG_CANT_START
+	}
+
+	appErrorChannel := make(chan error, 1)
+
+	changes <- svc.Status{State: svc.StartPending}
+	if err = s.start(elog, appErrorChannel); err != nil {
+		elog.Error(3, fmt.Sprintf("failed to start service: %v", err))
+		return true, 1064 // 1064: ERROR_EXCEPTION_IN_SERVICE
```

Comment:
`true` here means that the return code is not standard, though it seems that it is https://docs.microsoft.com/en-us/dotnet/api/microsoft.hpc.scheduler.properties.errorcode.error_exception_in_service?view=hpc-sdk-5.1.6115 ?
Possibly I misunderstand the purpose of the first flag.

Reply:
👍 - I was playing around with seeing how the errors occured and must have forgot to change this back. I've updated this to return `false` everywhere now.

Note even though the docs say this boolean means it's a service specfic error code, it doesn't seem to impact how its shown in Event Viewer.